store/principal.go

```diff
@@ -206,7 +206,7 @@ func (s *Store) findPrincipalRaw(ctx context.Context, find *api.PrincipalFind) (
 	}
 
 	if len(list) == 0 {
-		return nil, &common.Error{Code: common.NotFound, Err: fmt.Errorf("principal not found with PrincipalFind[%+v]", find)}
+		return nil, nil
```

Comment:
is this an intentional change?

@dragonly FYI

Reply:
Yes. Before #971 , this line will return nil instead of an error; and I have no clue why this line was changed to return an NotFound error. 
I have already left some comments at #971.

Reply:
You may refer to #307, where @d-bytebase refactored this line to let it return nil instead of an error.

We had an off-line discussion about this refactoration before. In a nutshell, returning nil, rather than an error, have the following advantages:
1. nil can represent `Empty / Not Found` semantic better
2. It is alwasy favorable to check whether a pointer is nil before using it.
3. User may not need to parse the Error to determine whther it is an NotFound Error.

Reply:
Please take a look at this file @dragonly .

Reply:
Yes, we can do refactoring without changing the current behavior. IMO, the best way to handle this is to return a tuple (value, ok, error) similar to getting a value from a map such as v, ok = m[k].

Reply:
I'll also check other PRs in the stage 2 refactoring of #758, and revert them to return `nil, nil`.

Reply:
> Yes, we can do refactoring without changing the current behavior. IMO, the best way to handle this is to return a tuple (value, ok, error) where the value is not a pointer, similar to getting a value from a map such as v, ok = m[k].

Maybe we could address this through another PR.